### ISSUE TITLE
Support hmset with an array

### ISSUE THREAD

**Issue by vipinj** [brigade/mock_redis#126]
```
 require 'mock_redis'
 => true

REDIS1 = MockRedis.new
 => #<MockRedis:0x007feccd959e28 @options={:scheme=>"redis", :host=>"127.0.0.1", :port=>6379, :path=>nil, :timeout=>5.0, :password=>nil, :db=>0, :time_class=>Time}, @db=#<MockRedis::PipelinedWrapper:0x007feccd951ed0 @db=#<MockRedis::TransactionWrapper:0x007feccd9523d0 @db=#<MockRedis::ExpireWrapper:0x007feccd952448 @db=#<MockRedis::MultiDbWrapper:0x007feccd958c80 @db_index=0, @prototype_db=#<MockRedis::Database:0x007feccd958870 @base=#<MockRedis:0x007feccd959e28 ...>, @data={}, @expire_times=[]>, @databases={0=>#<MockRedis::Database:0x007feccd959360 @base=#<MockRedis:0x007feccd959e28 ...>, @data={}, @expire_times=[]>}>>, @transaction_futures=[], @in_multi=false, @multi_block_given=false>, @pipelined_futures=[], @in_pipeline=false>>

a = {1 => 2, 3=>4}.to_a.flatten
 => [1, 2, 3, 4]

 REDIS1.hmset('b',a)
Redis::CommandError: ERR wrong number of arguments for HMSET
	from /Users/vjain/.rvm/gems/ruby-2.2.3@subject_data_integration/gems/mock_redis-0.16.0/lib/mock_redis/hash_methods.rb:93:in `hmset'
	from /Users/vjain/.rvm/gems/ruby-2.2.3@subject_data_integration/gems/mock_redis-0.16.0/lib/mock_redis/multi_db_wrapper.rb:21:in `method_missing'
	from /Users/vjain/.rvm/gems/ruby-2.2.3@subject_data_integration/gems/mock_redis-0.16.0/lib/mock_redis/expire_wrapper.rb:17:in `method_missing'
	from /Users/vjain/.rvm/gems/ruby-2.2.3@subject_data_integration/gems/mock_redis-0.16.0/lib/mock_redis/transaction_wrapper.rb:30:in `method_missing'
	from /Users/vjain/.rvm/gems/ruby-2.2.3@subject_data_integration/gems/mock_redis-0.16.0/lib/mock_redis/pipelined_wrapper.rb:27:in `method_missing'
	from /Users/vjain/.rvm/gems/ruby-2.2.3@subject_data_integration/gems/mock_redis-0.16.0/lib/mock_redis.rb:89:in `method_missing'
	from (irb):4
	from /Users/vjain/.rvm/gems/ruby-2.2.3@subject_data_integration/gems/railties-4.2.5.2/lib/rails/commands/console.rb:110:in `start'
	from /Users/vjain/.rvm/gems/ruby-2.2.3@subject_data_integration/gems/railties-4.2.5.2/lib/rails/commands/console.rb:9:in `start'
	from /Users/vjain/.rvm/gems/ruby-2.2.3@subject_data_integration/gems/railties-4.2.5.2/lib/rails/commands/commands_tasks.rb:68:in `console'
	from /Users/vjain/.rvm/gems/ruby-2.2.3@subject_data_integration/gems/railties-4.2.5.2/lib/rails/commands/commands_tasks.rb:39:in `run_command!'
	from /Users/vjain/.rvm/gems/ruby-2.2.3@subject_data_integration/gems/railties-4.2.5.2/lib/rails/commands.rb:17:in `<top (required)>'
	from bin/rails:4:in `require'
	from bin/rails:4:in `<main>'


REDIS2 = Redis.new(url: 'redis://localhost:6379')
 => #<Redis client v3.2.1 for redis://localhost:6379/0>
 REDIS2.hmset('b',a)
 => "OK"
```

**Comment by trotzig** [comment]
Thanks for the report @vipinj. This was fixed in dfbd027a479 and released in `0.17.1` a while ago.

**Comment by vipinj** [comment]
nice, thanks.
